src/v/cloud_storage/partition_manifest.h

```diff
@@ -337,7 +337,10 @@ class partition_manifest : public base_manifest {
     bool add(const segment_name& name, const segment_meta& meta);
 
     /// Return 'true' if the segment meta can be added safely
-    bool safe_segment_meta_to_add(const segment_meta& meta);
+    bool safe_segment_meta_to_add(const segment_meta& meta) const;
+    /// Return number of segments (counting from the beginning of the list) that
+    /// can be added safely.
+    size_t safe_segment_meta_to_add(std::vector<segment_meta> list) const;
```

Comment:
nit: can be const ref to avoid copying on each invocation?

Reply:
I'll double check that the value is moved and not copied.